### PR TITLE
Cover UI improvements

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: src/modules/jcms_admin/jcms_admin.module [hunk 1]
```diff
@@ -128,6 +159,32 @@ function jcms_admin_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $for
...
+ * @param string $key
+ * @param mixed|NULL $value
+ * @return mixed|NULL
+ */
+function _jcms_admin_covers_active($key, $value = NULL) {
+  static $values = [];
+  if (!is_null($value)) {
+    $values[$key] = $value;
+  }
+  return (isset($values[$key])) ? $values[$key] : $value;
```
Comment (jamiehollern) [inline src/modules/jcms_admin/jcms_admin.module]: `return $values[$key] ?? $value;`

File: src/modules/jcms_admin/jcms_admin.module [hunk 2]
```diff
@@ -193,3 +250,55 @@ function _jcms_admin_node_title_unique($title, $node, $content_type = NULL) {
...
+  if (_jcms_admin_covers_active('publish')) {
+    $items = ($entity->get('items')) ? $entity->get('items')->getValue() : [];
+    $items = array_splice($items, 0, $entity->get('field_covers_active_items')->getString());
+    foreach ($items as $item) {
+      $node = Node::load($item['target_id']);
+      $moderation_info = \Drupal::service('content_moderation.moderation_information');
+      if (!$moderation_info->isLatestRevision($node)) {
+        $latest = $moderation_info->getLatestRevision($node->getEntityTypeId(), $node->id());
+        $latest->moderation_state->target_id = 'published';
+        $latest->save();
```
Comment (jamiehollern) [inline src/modules/jcms_admin/jcms_admin.module]: Should you be executing this and the `$node->save()` below in a presave hook?

File: src/modules/jcms_admin/jcms_admin.module [hunk 3]
```diff
@@ -193,3 +250,55 @@ function _jcms_admin_node_title_unique($title, $node, $content_type = NULL) {
...
+      ]));
+    $subqueue->save();
+  }
+  if (_jcms_admin_covers_active('reset')) {
+    $subqueue = EntitySubqueue::load('covers');
+    $items = ($subqueue->get('items')) ? $subqueue->get('items')->getValue() : [];
+    $entity->set('field_covers_active_items', 3);
+    foreach ($items as $item) {
+      $node = Node::load($item['target_id']);
+      $node->moderation_state->target_id = 'published';
```
Comment (jamiehollern) [inline src/modules/jcms_admin/jcms_admin.module]: Maybe store this string as a constant.

File: src/modules/jcms_rest/src/Plugin/rest/resource/AbstractRestResourceBase.php [hunk 4]
```diff
@@ -575,4 +576,16 @@ public function getContentType(int $version = 1): string {
     return $content_type . ';version=' . $version;
   }
 
+  public function viewUnpublished() {
+    $request = \Drupal::request();
+    if ($username = $request->getUser()) {
+      /* @var User $user */
+      $user = user_load_by_name($username);
+      return $user->hasPermission('view latest version');
```
Comment (jamiehollern) [inline src/modules/jcms_rest/src/Plugin/rest/resource/AbstractRestResourceBase.php]: `return \Drupal::currentUser()->hasPermission('view latest version')`;

File: src/modules/jcms_rest/src/Plugin/rest/resource/CoverCurrentListRestResource.php [hunk 5]
```diff
@@ -28,8 +28,26 @@ class CoverCurrentListRestResource extends AbstractRestResourceBase {
...
    */
   public function get() {
+    if ($this->viewUnpublished()) {
+      $response_data = $this->getPreview();
+    }
+    else {
+      $response_data = $this->getPublished();
+    }
+
+    $response = new JsonResponse($response_data, Response::HTTP_OK, ['Content-Type' => 'application/vnd.elife.cover-list+json;version=1']);
```
Comment (jamiehollern) [inline src/modules/jcms_rest/src/Plugin/rest/resource/CoverCurrentListRestResource.php]: `'Content-Type' => $this->getContentType()`

File: src/modules/jcms_admin/jcms_admin.module (hunk 2 above)
Comment (nlisgo) [inline src/modules/jcms_admin/jcms_admin.module]: `$latest` and `$node` refer to the items in the `entity_subqueue`. The presave hook is on the `entity_aubqueue` so I believe the are necessary.

File: src/modules/jcms_admin/jcms_admin.module (hunk 3 above)
Comment (nlisgo) [inline src/modules/jcms_admin/jcms_admin.module]: I looked in the `content_moderation` module for a constant for this value and couldn't find one. I think it's a good idea but it really should belong in that module.

**Pull request merged**